libraries/chain/controller.cpp

```diff
@@ -1990,6 +1994,16 @@ struct controller_impl {
       emit( self.new_hs_new_block_message, msg );
    }
 
+   void set_finalizers_impl(uint64_t fthreshold, vector<finalizer_authority> finalizers) {
+      this->fthreshold = fthreshold;
+      this->finalizers = finalizers;
+   }
+
+   void get_finalizers_impl(uint64_t& fthreshold, vector<finalizer_authority>& finalizers) {
+      fthreshold = this->fthreshold;
+      finalizers = this->finalizers;
+   }
```

Comment:
I think it would be nicer to have:
```
std::pair<uint64_t, vector<finalizer_authority>> get_finalizers_impl() {
   return { fthreshold, finalizers );
}
```

which you can the use as:

```
auto [threshold, finalizers] = controller.get_finalizers();
```